support/android/apk/app/src/main/java/com/mozilla/servo/MainActivity.java

```diff
@@ -0,0 +1,303 @@
+        System.loadLibrary("c++_shared");
+        System.loadLibrary("servo");
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        try {
+            extractAssets();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        JSONObject preferences = loadPreferences();
+        boolean keepScreenOn = false;
+        String orientation = "both";
+        if (preferences != null) {
+            keepScreenOn = preferences.optBoolean("shell.keep_screen_on.enabled", false);
+            mFullScreen = !preferences.optBoolean("shell.native-titlebar.enabled", false);
+            orientation = preferences.optString("shell.native-orientation", "both");
+        }
+
+        // Handle orientation preference
+        if (orientation.equalsIgnoreCase("portrait")) {
+            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+        }
+        else if (orientation.equalsIgnoreCase("landscape")) {
+            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+        }
+
+        super.onCreate(savedInstanceState);
+
+        // Handle keep screen on preference
+        if (keepScreenOn) {
+            keepScreenOn();
+        }
+
+        // Handle full screen preference
+        if (mFullScreen) {
+            addFullScreenListener(this.getWindow().getDecorView());
+        }
+        
+        final Intent intent = getIntent();
+        if (intent != null && intent.getAction().equals(Intent.ACTION_VIEW)) {
+            final String url = intent.getDataString();
+            if (url != null && URLUtil.isValidUrl(url)) {
+                Log.d(LOGTAG, "Received url "+url);
+                set_url(url);
+            }
+        }
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();  // Always call the superclass method first
+        Log.d(LOGTAG, "onStop");
+    }
+
+    @Override
+    protected void onPause() {
+        if (mWakeLock != null) {
+            mWakeLock.release();
+        }
+        super.onPause();
+        Log.d(LOGTAG, "onPause");
+    }
+
+    @Override
+    protected void onResume() {
+        if (mFullScreen) {
+            setFullScreen();
+        }
+        if (mWakeLock != null) {
+            mWakeLock.acquire();
+        }
+
+        super.onResume();
+        Log.d(LOGTAG, "onPause");
+    }
+
+    // keep the device's screen turned on and bright.
+    private void keepScreenOn() {
+        if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR1 ) {
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+        }
+        else {
+            final PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
+            mWakeLock = pm.newWakeLock(PowerManager.SCREEN_DIM_WAKE_LOCK, getClass().getName());
+        }
+    }
+
+    // Dim toolbar and make the view fullscreen
+    private void setFullScreen()
+    {
```

Comment:
Same here.

Also the style in this file is not consistent. Either always put the opening brace on its own line or on the same line as the method declaration, but don't mix and match.